python/lsst/daf/persistence/butler.py

```diff
@@ -369,13 +360,17 @@ def subset(self, datasetType, level=None, dataId={}, **rest):
         @returns (ButlerSubset) collection of ButlerDataRefs for datasets
         matching the data id.
         """
-
         datasetType = self._resolveDatasetTypeAlias(datasetType)
         if level is None:
-            level = self.mapper.getDefaultLevel()
-        dataId = self._combineDicts(dataId, **rest)
+            # todo this does not seem safe. what if repo is aggregate, and conained mappers have different
+            # levels? ...it seems wrapped up in the problem with ButlerSubset.
```

Comment:
maybe the repository.levels call should happen in ButlerSubset.
I forget what "the problem with ButlerSubset is". Will look into it.
remove commentage.